data/pools/transactionPool.go

```diff
@@ -187,24 +201,34 @@ func (pool *TransactionPool) test(t transactions.SignedTxn) error {
 		feePerByte = feePerByte << (pool.numPendingWholeBlocks - 1)
 	}
 
-	feeThreshold := feePerByte * uint64(t.GetEncodedLength())
-	if t.Txn.Fee.Raw < feeThreshold {
-		return fmt.Errorf("fee %d below threshold %d (%d per byte * %d bytes)",
-			t.Txn.Fee, feeThreshold, feePerByte, t.GetEncodedLength())
+	for _, t := range txgroup {
+		feeThreshold := feePerByte * uint64(t.GetEncodedLength())
+		if t.Txn.Fee.Raw < feeThreshold {
+			return fmt.Errorf("fee %d below threshold %d (%d per byte * %d bytes)",
+				t.Txn.Fee, feeThreshold, feePerByte, t.GetEncodedLength())
+		}
 	}
 
 	return nil
 }
 
-// Remember stores the provided transaction
+// RememberOne stores the provided transaction
+// Precondition: Only RememberOne() properly-signed and well-formed transactions (i.e., ensure t.WellFormed())
+func (pool *TransactionPool) RememberOne(t transactions.SignedTxn) error {
+	return pool.Remember([]transactions.SignedTxn{t})
+}
+
+// Remember stores the provided transaction group
 // Precondition: Only Remember() properly-signed and well-formed transactions (i.e., ensure t.WellFormed())
```

Comment:
Not related to this PR's changes, but do you know if this precondition (check WellFormed before calling Remember) is still necessary?

Reply:
That comment is certainly misleading.  The well-formedness doesn't matter much.  It's critical that the transaction signature has been checked, though.  The pool assumes the signature is already valid (doesn't do its own check), and the block evaluator subsequently assumes any transactions coming from the pool have good signatures.